lib/factories/default-eventbus.js

```diff
@@ -0,0 +1,373 @@
+                    `Event must have a '${schemaUriField}' property`
+                );
+            }
+        }
+    };
+}
+
+/**
+ * Returns a function that given a URI, will resolve it using
+ * options.schema_base_uri and options.schema_file_extension.
+ * See also the event-utils resolveUri function.
+ * @param {Object} options
+ * @param {string} options.schema_base_uri
+ *      If set, this is prefixed to un-anchored schema URIs.
+ * @param {string} options.schema_file_extension
+ *      If set, this is suffixed to schema URIs that dont' already have a file extension.
+ * @return {function(string): string}
+ */
+function makeResolveSchemaUri(options) {
+    return uri => resolveUri(
+        uri, options.schema_base_uri, options.schema_file_extension
+    );
+}
+
+/**
+ * If options.stream_field is configured, then it will be used to extract
+ * the event's destination stream name.  Otherwise, it is assumed
+ * that the event's sanitized schema URI should be used as the stream name.
+ * The schema URI returned by the function returned by makeExtractSchemaUri
+ * with options.schema_uri_field will be used to extract the schema URI,
+ * and then the schema URI will be sanitized to remove potential bad charachters.
+ * Any character that does not match [A-Za-z0-9_-] will be replaced with
+ * an underscore.
+ * @param {Object} options
+ * @param {stirng} options.stream_field
+ *      If given, used to extract the event's destination stream name.
+ * @param {string} options.schema_uri_field
+ *      Used to extract the event's schema URI to use as a sanitized stream name.
+ *      This is only used if options.stream_field is not given.
+ */
+function makeExtractStreamName(options) {
+    if (_.isUndefined(options.stream_field)) {
+        const extractSchemaUri = makeExtractSchemaUri(options);
+        const badCharsRegex = new RegExp('[^A-Za-z0-9_-]', 'g');
+        const replacementChar = '_';
+        return (event) => {
+            const schemaUri = extractSchemaUri(event);
+            return schemaUri.replace(badCharsRegex, replacementChar);
+        };
+    } else {
+        return (event) => {
+            return objectGet(event, options.stream_field);
+        };
+    }
+}
+
+/**
+ * Creates a function that extracts a topic (and potentially adds a prefix)
+ * using options.stream_field and options.topic_prefix.
+ * @param {Object} options
+ * @param {stirng} options.stream_field
+ *      Used to extract the event's destination stream name.
+ * @param {stirng} options.topic_prefix
+ *      If given, this will be prefixed to the value extracted from stream_field
+ *      and used as the topic in Kafka.
+ * @return {function(Object): string}
+ */
+function makeExtractTopic(options) {
+    const extractStreamName = makeExtractStreamName(options);
+    return (event) => {
+        const streamName = extractStreamName(event);
+        return options.topic_prefix ?  options.topic_prefix + streamName : streamName;
+    };
```

Comment:
This makes me think whether it's worth making some kind of a simple templating language for configuring all of the names. Probably now..

Reply:
Eh?  Whatcha mean?